GridList/GridList.jsx

```diff
@@ -0,0 +1,151 @@
+    };
+  }
+
+  isValidValue(validationValues, testValue) {
+    return (
+      validationValues &&
+      validationValues.findIndex(val => val === testValue) >= 0
+    );
+  }
+
+  constructor() {
+    super();
+    this.componentName = "grid-list";
+    this._mdcProps = [
+      "header-caption",
+      "twoline-caption",
+      "tile-gutter",
+      "with-icon-align",
+      "tile-aspect-ratio"
+    ];
+  }
+  mapClassName(propKey, props) {
+    const propValue = props[propKey];
+    const validationValues = this.validationValuesByKey[propKey];
+
+    return this.isValidValue(validationValues, propValue)
+      ? `mdc-${this.componentName}--${propKey}-${propValue}`
+      : "";
+  }
+  materialDom(props) {
+    const className = Object.keys(this.validationValuesByKey)
+      .map(key => {
+        return this.mapClassName(key, props);
+      })
+      .filter(notEmptyString)
+      .join(" ");
+
+    return (
+      <div {...props} className={className}>
+        {props.children}
+      </div>
+    );
+  }
+}
+
+class GridListTiles extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list__tiles";
+  }
+  materialDom(props) {
+    return <ul {...props}>{props.children}</ul>;
+  }
+}
+
+class GridListTile extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile";
+  }
+  materialDom(props) {
+    return <li {...props}>{props.children}</li>;
+  }
+}
+
+class GridListTilePrimary extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__primary";
+  }
+  materialDom(props) {
+    return <div {...props}>{props.children}</div>;
+  }
+}
+
+class GridListTilePrimaryContent extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__primary-content";
+  }
+  materialDom(props) {
+    return <img {...props} />;
+  }
+}
+
+class GridListTileSecondary extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__secondary";
+  }
+  materialDom(props) {
+    return <span {...props}>{props.children}</span>;
+  }
+}
+
+class GridListTileTitle extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__title";
+  }
+  materialDom(props) {
+    return <span {...props}>{props.children}</span>;
+  }
+}
+
+class GridListTileSupportText extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__support-text";
+  }
+  materialDom(props) {
+    return <span {...props}>{props.children}</span>;
+  }
+}
+
+class GridListTileIcon extends Icon {
+  constructor() {
+    super();
+    this.componentName = "grid-tile__icon";
+  }
+}
+
+GridList.Tiles = GridListTiles;
+GridList.Tile = GridListTile;
+GridList.Tile.Primary = GridListTilePrimary;
+GridList.Tile.Primary.Content = GridListTilePrimaryContent;
+GridList.Tile.Secondary = GridListTileSecondary;
```

Comment:
suggestion: `Grid.SecondaryTile` and so on for other components too

Reply:
To me this loses the implied structural hierarchy in the names (a GridTile and a GridIconTile sound like different flavors of the same thing rather than the GridIconTile needing to be a child of the GridTile), but I've renamed them per your suggestion.